samples/net/sntp/src/main.c

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2024 Embeint Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+ #include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(net_sntp_sample, LOG_LEVEL_DBG);
+
+#include <time.h>
+
+#include <zephyr/net/socket.h>
+#include <zephyr/net/socket_service.h>
+#include <zephyr/net/sntp.h>
+
+#include "net_sample_common.h"
+
+static K_SEM_DEFINE(sntp_async_received, 0, 1);
+static void sntp_service_handler(struct net_socket_service_event *pev);
+
+NET_SOCKET_SERVICE_SYNC_DEFINE_STATIC(service_sntp_async, sntp_service_handler, 1);
+
+int dns_query(const char *host, uint16_t port, int family, int socktype, struct sockaddr *addr,
+	      socklen_t *addrlen)
+{
+	struct zsock_addrinfo hints = {
+		.ai_family = family,
+		.ai_socktype = socktype,
+	};
+	struct zsock_addrinfo *res = NULL;
+	int rc;
+
+	/* Perform DNS query */
+	rc = zsock_getaddrinfo(host, NULL, &hints, &res);
+	if (rc < 0) {
+		LOG_ERR("zsock_getaddrinfo failed (%d, errno %d)", rc, errno);
+		return rc;
+	}
+	/* Store the first result */
+	*addr = *res->ai_addr;
+	*addrlen = res->ai_addrlen;
+	/* Free the allocated memory */
+	zsock_freeaddrinfo(res);
+
+	if (addr->sa_family == AF_INET) {
+		struct sockaddr_in *ipv4 = (struct sockaddr_in *)addr;
+		uint8_t *b = ipv4->sin_addr.s4_addr;
+
+		ipv4->sin_port = htons(port);
+		LOG_INF("%s -> %d.%d.%d.%d:%d", host, b[0], b[1], b[2], b[3], port);
+	} else {
+		struct sockaddr_in6 *ipv6 = (struct sockaddr_in6 *)addr;
+
+		ipv6->sin6_port = htons(port);
+		LOG_INF("%s -> IPv6:%d", host, port);
+	}
```

Comment:
This looks a bit of a kludge, I mean there is `inet_ntop` that can create the printable ip address, and we don't even need this if()else thingy.

Reply:
That function exists in `<zephyr/posix/arpa/inet.h>`, is there any expectation that the function will exist on non-posix targets?